docs/tileentities/tileentity.md

```diff
@@ -0,0 +1,128 @@
+```JAVA
+    Block#hasTileEntity(IBlockstate state)
+
+    Block#createTileEntity(World world, IBlockState state)
+```
+Using the parameters you can choose if the block should have a `TileEntity` or not.
+But usually you will return `true` in the first method and a new instance of your `TileEntity` in the second method.
+
+At this point the `TileEntity`can be used for Rendering.
+
+## Storing Data within your `TileEntity`
+In order to save data, override the following two methods
+```JAVA
+    TileEntity#writeToNBT(NBTTagCompound nbt)
+
+    TileEntity#readFromNBT(NBTTagCompound nbt)
+```
+These methods are called whenever the `TileEntity` gets (un-)loaded.
+Use them to read and write to the fields in your tile entity class.
+
+!!! warning
+    Whenever your data changes you need to call ```TileEntity#markDirty()```, otherwise your `TileEntity` might be skipped while saving.
+
+!!! note
+    It is important that you call the super methods!
+    The tag names `id`, `x`, `y`, `z`, `ForgeData` and `ForgeCaps`tug are reserved by the super methods.
+
+## Keeping a `TileEntity` through changing `BlockStates`
+There might be situations in which you need to change your `BlockState`, an example for this is the vanilla furnace,
+which changes its state from `lit=false` to `lit=true` when fuel and something smetlable is inside.
+Achieving this is rather simple, by overriding following method
+```JAVA
+    TileEntity#shouldRefresh(World world, BlockPos pos, IBlockState oldState, IBlockState newSate)#boolean
+```
+!!! note
+    You should actually check the `BlockStates` and not just return `false` in order to prevent unwanted behavior and bugs.
+    
+## Ticking `TileEntities`
+If you need a ticking `TileEntity` for example to keep track of the progress during a smelting process.
+You need to add the `ITickable` interface to your `TileEntity`.
+Now you can implement all your calculations within
+```JAVA
+    ITickable#update()
+```
+!!! note
+    This method is called each tick, therefore you should avoid having complicated calculations in here.
+    If possible, you should make more complex calculations just every X ticks.
+    (The amount of ticks in a second may be lower then 20 (twenty) but won't be higher)
+    
+## Synchronizing the Data to the Client
+There are 3 (three) ways of syncing data to the client.
+Synchronizing on chunk load, synchronizing on block updates and synchronizing with a custom network message.
+
+### Synchronizing on chunk load
+For this you need to override
+```JAVA
+    TileEntity#getUpdateTag()
+
+    TileEntity#handleUpdateTag(NBTTagCompound nbt)
+```
+Again, this is pretty simple, the first method collects the data that should be send to the client.
+While the second one processes that data. If your `TileEntity` doesn't contain much data you might be able to use the methods out of the `Storing Data within your TileEntity` section.
```

Comment:
`... the first method collects the data that should be send to the client. While the second one processes that data.` The second sentence here doesn't work on its own.